cmd/clusterctl/api/v1alpha3/provider_type.go

```diff
@@ -95,11 +83,19 @@ func (p *Provider) GetProviderType() ProviderType {
 	}
 }
 
+// InstanceName return the instance name for the provider, that is composed by the provider name and the namespace
+// where the provider is installed (nb. clusterctl does not support more instance of the same provider to be
+// installed in the same namespace)
+func (p *Provider) InstanceName() string {
```

Comment:
Adding instanceName so I can drop the usage of name/namespace tuple in many places